health-services/ExerciseSampleCompose/app/src/main/java/com/example/exercisesamplecompose/presentation/goals/ExerciseGoalsViewModel.kt

```diff
@@ -0,0 +1,38 @@
+package com.example.exercisesamplecompose.presentation.goals
+
+import androidx.compose.runtime.State
+import androidx.compose.runtime.mutableStateOf
+import androidx.lifecycle.SavedStateHandle
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import com.example.exercisesamplecompose.data.ExerciseClientManager
+import dagger.hilt.android.lifecycle.HiltViewModel
+import kotlinx.coroutines.launch
+import javax.inject.Inject
+import androidx.compose.runtime.asDoubleState
+import androidx.compose.runtime.asLongState
+import androidx.compose.runtime.mutableStateOf
+
+
+
+@HiltViewModel
+class ExerciseGoalsViewModel @Inject constructor(
+    savedStateHandle: SavedStateHandle,
+    private val exerciseClientManager: ExerciseClientManager
+) : ViewModel() {
+
+    private val _distanceGoal = mutableStateOf(0.0)
+    val distanceGoal: State<Double> = _distanceGoal.asDoubleState()
+
+    private val _durationGoal = mutableStateOf(0L)
+    val durationGoal: State<Long> = _durationGoal.asLongState()
+
+
+    // Set new goals and update the values in ExerciseClientManager
+    fun setGoals(newDistance: Double, newDuration: Long) {
```

Comment:
As above, combine distance and duration into one data class. This then means that anything that is observing this state will only get one change, whereas at the moment, distance and duration goals are changing two separate states, so potentially the UI is getting refreshed twice? (maybe?)

Reply:
Done!